[MTM-56477] User documentation for time series migration process (#2478)
SoftwareAG/c8y-docs: content/enterprise-tenant/enhanced-time-series-support.md

---
title: Enhanced time series support
layout: bundle
outputs:
  - html
  - json
sector:
  - platform_administration

> weight: 11

Reviewer: ```suggestion
weight: 80
```

```

If the configuration is not set for the tenant at all, you will get a 404 response code for the request above.

### To trigger time series migration {#to-trigger-time-series-migration}

The administrator user can trigger the migration to time series collection via user interface of a plugin <provide the name of the plugin>, which is installed in the Administration application by default. Optionally, the plugin can be installed in other applications on {{< management-tenant >}} or {{< enterprise-tenant >}}.

> The plugin provides an opportunity to migrate to time series collection for the current tenant ({{< management-tenant >}} or {{< enterprise-tenant >}}) and for the list of subtenants, where you can select a specific tenant for the migration.

Reviewer: ```suggestion
The plugin provides an opportunity to migrate to time series collection for the current ({{< management-tenant >}} or {{< enterprise-tenant >}}) and for the list of subtenants, where you can select a specific tenant for the migration.
```

```

If the configuration is not set for the tenant at all, you will get a 404 response code for the request above.

> 

Reviewer 2: I'd have a comment on the context help content at the top:
```
and provide the option to remove legacy collection afterwards.
```
I wouldn't call it an "option", because once migration is approved, the legacy data will be deleted anyway. What do you think?

Author: I've changed to `Migrated tenants require user approval and provide the opportunity to remove legacy collection afterwards.` is it ok?

Reviewer 2: hmm, no, because the removal is not optional, it will happen after the scheduled time. Maybe: `Migrated tenants require user approval and their legacy measurement collections are removed after seven days.`?

Reviewer 2: I see it's applied, so resolving thread.

### To trigger time series migration {#to-trigger-time-series-migration}

> The admin user can trigger time series measurements migration via user interface using the plugin. The plugin should be installed on management tenant or on enterprise tenant.

Reviewer 3: ```suggestion
The administrator user can trigger the time series measurements migration via user interface using the plugin. The plugin should be installed on {{< management-tenant >}} or on {{< enterprise-tenant >}}.
```

> The administrator user can trigger the time series measurements migration via user interface using the plugin. The plugin should be installed on {{< management-tenant >}} or on {{< enterprise-tenant >}}.

Reviewer 2: ```suggestion
The administrator user can trigger the migration to time series collection via user interface of a plugin <provide the name of the plugin>, which is installed in the Administration application by default. Optionally, the plugin can be installed in other applications on {{< management-tenant >}} or {{< enterprise-tenant >}}.
```

> The plugin provides opportunity to migrate time series collection for current tenant (management or enterprise tenant) and for the list of subtenants, where user can select specific tenant for migration.

Reviewer 3: ```suggestion
The plugin provides opportunity to migrate the time series collection for the current tenant ({{< management-tenant >}} or{{< enterprise-tenant >}} ) and for the list of subtenants, where you can select a specific tenant for the migration.
```

> To start tenant migration follow these steps:

Reviewer 3: ```suggestion
To start the tenant migration follow the steps below:
```

> The plugin provides opportunity to migrate the time series collection for the current tenant ({{< management-tenant >}} or{{< enterprise-tenant >}} ) and for the list of subtenants, where you can select a specific tenant for the migration.

Reviewer 2: ```suggestion
The plugin provides an opportunity to migrate to time series collection for the current tenant ({{< management-tenant >}} or {{< enterprise-tenant >}}) and for the list of subtenants, where you can select a specific tenant for the migration.
```

> 1. Navigate to **Migration > Time series** in application.

Reviewer 3: @darluk In which application?

Author: @MWindel  actually the plugin can be installed in any application, so maybe it worth to write `Navigate to **Migration > Time series** in application, where plugin is installed.` ?

Reviewer 2: @darluk, sounds good but maybe we could also add info about Administration (we will be installing it there by default), e.g.:
```suggestion
1. Navigate to **Migration > Time series** in the application where the plugin is installed (Administration by default).
```

> 2. Select the tenant you want to trigger migration for from the list of available tenants.

Reviewer 3: ```suggestion
2. Select the tenant you want to trigger the migration for from the list of available tenants.
```

> 3. Hover over on the row of tenant in the tenant list, then click appeared button **Add to queue** and confirm operation. Tenant migration status should be updated to *Queued*.

Reviewer 3: ```suggestion
3. Hover over the row of the tenant in the tenant list, then click **Add to queue** and confirm the operation. The tenant migration status should be updated to **Queued**.
```

> 4. When migration process will be triggered its status for the tenant should be changed from *Queued* to *In progress*.

Reviewer 3: ```suggestion
4. When the migration process is triggered its status for the tenant should changed from **Queued** to **In progress**.
```

Reviewer 2: if we drop "be", then "should change"

> After the data will be processed, verified and migrated to the new collection the user can confirm progress by clicking **Approve and finish migration**.

Reviewer 3: ```suggestion
After the data is processed, verified and migrated to the new collection, confirm progress and click **Approve and finish migration**.
```

Reviewer 2: verified before migration?
```suggestion
After the data is migrated to the new collection and verified from the technical perspective, confirm the migration result and click **Approve and finish migration**.
```
Maybe we need more details how user can make sure the migration worked fine for their data? Maybe put some points from the dialog that we display there?

> 5. Confirmation pop-up will appear to inform the user that the new format for time series measurements is going to be used after confirming the migration.

Reviewer 3: ```suggestion
5. A confirmation pop-up will appear to show that the new format for the time series measurements is going to be used after confirming the migration. Click **Accept**.
```

Reviewer 2: The pop up also mentions other points, right? Might be good to elaborate on those points so that user is better prepared what to expect before they get to this dialog in runtime. I think the label for the confirmation button in the dialog is "Confirm", not "Accept".

> 6. Accepting confirmation pop-up changes status of the migration to *Approved*.

Reviewer 3: ```suggestion
6. Accepting the confirmation pop-up changes the status of the migration to **Approved**.
```

> 7. After 7 days migration will be fully finished and its status for selected tenant will be changed to *Completed*, afterwards old time series collection will be removed.

Reviewer 3: ```suggestion
7. After seven days, the migration will be fully finished and its status for the selected tenant is changed to **Completed**. Afterwards, th old time series collection will be removed.
```

> 7. After seven days, the migration will be fully finished and its status for the selected tenant is changed to **Completed**. Afterwards, th old time series collection will be removed.

Reviewer 2: ```suggestion
7. After seven days, the legacy measurements collection will be deleted and the migration status will be changed to **Completed**.
```

{{< c8y-admon-info >}}

> Migration of measurements can be cancelled when tenant has *Queued* status. After status changed to *In progress* the process can no longer be stopped. However if the user will not approve migration manually then migration stays in status *Verified*.

Reviewer 3: ```suggestion
Migration of measurements can be cancelled when a tenant has the status **Queued**. After the status changed to **In progress**, the process can no longer be stopped. However, if you will not approve the migration manually, the migration stays in the status **Verified**.
```

> Migration of measurements can be cancelled when a tenant has the status **Queued**. After the status changed to **In progress**, the process can no longer be stopped. However, if you will not approve the migration manually, the migration stays in the status **Verified**.

Reviewer 2: ```suggestion
The migration of measurements can be cancelled while the tenant has the status **Queued**. After the status is changed to **In progress**, the process can no longer be stopped. However, if you don't approve the migration manually, it will stay in the status **Verified**.
```
Maybe we should add: "and our support may help you clarify any doubts or revert your tenant to the original collection format." (check with backend if we want to declare that)?

Author: BE propose to avoid such sentences to decrease potential issues based on misunderstanding

Reviewer 2: ok, then don't add that last sentence. Marlene suggested other changes to the main sentence, so let's go with them instead.

{{< /c8y-admon-info >}}

### Progress monitoring {#progress-monitoring}

> Once migration is initiated the user can monitor the progress in status bar in the section **Ongoing migration**. Where in percentage scale is presented the progression of already migrated data into the new collection regarding the left one.

Reviewer 3: ```suggestion
Once the migration is initiated, you can monitor the progress in status bar in the section **Ongoing migration**. Where you can see the percentage scale of the progression of already migrated data into the new collection regarding the left collection.
```

> In the section **Ongoing migration** should be shown information indicating the user who have started the migration and full migration time range (when the migration was started and is going to be finished)

Reviewer 3: ```suggestion
In the section **Ongoing migration** you can find information indicating who has started the migration as well as the full migration time range, that is when the migration was started and is going to be finished.
```

> In the section **Ongoing migration** you can find information indicating who has started the migration as well as the full migration time range, that is when the migration was started and is going to be finished.

Reviewer 2: "that is when the migration was started and is going to be finished" I'm afraid it could be interpreted as the start/end time of the migration process itself, while it's more "time range of data to be migrated"

Reviewer 4: what do you propose?

Reviewer 2: the proposal was included in another suggestion which is already applied and further commented by Marlene (https://github.com/Cumulocity-IoT/c8y-docs/pull/2478/files#diff-185d3024cb7a138b0e55cc98116f1f2b7099a06c4a2007c83308e3fb5db14ce6R30)

> Once the migration is initiated, you can monitor the progress in status bar in the section **Ongoing migration**. Where you can see the percentage scale of the progression of already migrated data into the new collection regarding the left collection.
> In the section **Ongoing migration** you can find information indicating who has started the migration as well as the full migration time range, that is when the migration was started and is going to be finished.

Reviewer 2: ```suggestion
When a migration is being executed, you can monitor its progress in the **Ongoing migration** section. Here, you can see: what tenant is being migrated, what time range of data will be migrated, who started the migration. On the right side, you can see the percentage indicator showing how much data has already been migrated to the new collection.
```

{{< c8y-admon-important >}}

> The user can add more than one tenant into migration queue, but migration will be done only for one tenant at a time. It means if there is a tenant migration with the status *Verified* until user approval, the migration will not update the status to *Complete*, hence the migration for other queued tenant will not be started

Reviewer 3: ```suggestion
You can add more than one tenant into migration queue, but migration will be done only for one tenant at a time. It means that if there is a tenant migration with the status **Verified** until you approve, the migration will not update the status to **Complete**, hence the migration for other queued tenant will not be started
```

> You can add more than one tenant into migration queue, but migration will be done only for one tenant at a time. It means that if there is a tenant migration with the status **Verified** until you approve, the migration will not update the status to **Complete**, hence the migration for other queued tenant will not be started

Reviewer 2: ```suggestion
You can add more than one tenant into the migration queue, but migration will be executed only for one tenant at a time. It means that we won't start the migration of the next tenant in the queue, until you approve the previously migrated tenant to transition from **Verified** to **Complete** status.
```
It might be worth mentioning that it may happen that we start a migration of a tenant which user cannot see (e.g. a subtenant of another enterprise tenant). From the current user's perspective, it will be perceived as if their queue is not being picked up for some time.
